app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx

```diff
@@ -0,0 +1,188 @@
+import { ShortNumber } from 'mastodon/components/short_number';
+import DropdownMenu from 'mastodon/containers/dropdown_menu_container';
+import { useIdentity } from 'mastodon/identity_context';
+import { PERMISSION_MANAGE_TAXONOMIES } from 'mastodon/permissions';
+import { useAppDispatch } from 'mastodon/store';
+
+const messages = defineMessages({
+  followHashtag: { id: 'hashtag.follow', defaultMessage: 'Follow hashtag' },
+  unfollowHashtag: {
+    id: 'hashtag.unfollow',
+    defaultMessage: 'Unfollow hashtag',
+  },
+  adminModeration: {
+    id: 'hashtag.admin_moderation',
+    defaultMessage: 'Open moderation interface for #{name}',
+  },
+});
+
+const usesRenderer = (displayNumber: React.ReactNode, pluralReady: number) => (
+  <FormattedMessage
+    id='hashtag.counter_by_uses'
+    defaultMessage='{count, plural, one {{counter} post} other {{counter} posts}}'
+    values={{
+      count: pluralReady,
+      counter: <strong>{displayNumber}</strong>,
+    }}
+  />
+);
+
+const peopleRenderer = (
+  displayNumber: React.ReactNode,
+  pluralReady: number,
+) => (
+  <FormattedMessage
+    id='hashtag.counter_by_accounts'
+    defaultMessage='{count, plural, one {{counter} participant} other {{counter} participants}}'
+    values={{
+      count: pluralReady,
+      counter: <strong>{displayNumber}</strong>,
+    }}
+  />
+);
+
+const usesTodayRenderer = (
+  displayNumber: React.ReactNode,
+  pluralReady: number,
+) => (
+  <FormattedMessage
+    id='hashtag.counter_by_uses_today'
+    defaultMessage='{count, plural, one {{counter} post} other {{counter} posts}} today'
+    values={{
+      count: pluralReady,
+      counter: <strong>{displayNumber}</strong>,
+    }}
+  />
+);
+
+export const HashtagHeader: React.FC<{
+  tagId: string;
+}> = ({ tagId }) => {
+  const intl = useIntl();
+  const { signedIn, permissions } = useIdentity();
+  const dispatch = useAppDispatch();
+  const [tag, setTag] = useState<ApiHashtagJSON>();
+
+  useEffect(() => {
+    void dispatch(fetchHashtag({ tagId })).then((result) => {
+      if (isFulfilled(result)) {
+        setTag(result.payload);
+      }
+
+      return '';
+    });
+  }, [dispatch, tagId, setTag]);
+
+  const menu = useMemo(() => {
+    const tmp = [];
+
+    if (
+      tag &&
+      signedIn &&
+      (permissions & PERMISSION_MANAGE_TAXONOMIES) ===
+        PERMISSION_MANAGE_TAXONOMIES
+    ) {
+      tmp.push({
+        text: intl.formatMessage(messages.adminModeration, { name: tag.id }),
+        href: `/admin/tags/${tag.id}`,
+      });
+    }
+
+    return tmp;
+  }, [signedIn, permissions, intl, tag]);
+
+  const handleFollow = useCallback(() => {
+    if (!signedIn || !tag) {
+      return;
+    }
+
+    if (tag.following) {
+      setTag((hashtag) => hashtag && { ...hashtag, following: false });
+
+      void dispatch(unfollowHashtag({ tagId })).then((result) => {
+        if (isFulfilled(result)) {
+          setTag(result.payload);
+        }
+
+        return '';
```

Comment:
That looks a bit weird. Why return an empty string?

Reply:
It's an [eslint rule](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/always-return.md). I don't like it either! But it's not harmful. I guess it makes it clearer you can't chain another `then` on.